cmd/csidriver/main.go

```diff
@@ -87,8 +88,13 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err := fs.MkdirAll(filepath.Join(csiOpts.RootDir, dtcsi.GarbageCollectionPath), 0770); err != nil {
-		log.Error(err, "unable to create garbage collector directory for CSI Driver")
+	access := metadata.SqliteAccess{}
+	if err := access.Setup(); err != nil {
+		log.Error(err, "failed to setup database storage for CSI Driver")
+		os.Exit(1)
+	}
+	if err := metadata.CheckStorageCorrectness(mgr.GetClient(), &access, log); err != nil {
+		log.Error(err, "failed to correct database storage for CSI Driver")
 		os.Exit(1)
 	}
```

Comment:
If housekeeping fails for whatever reason, the driver will terminate. This might result in an outage which is not automatically recoverable.